clarity/contracts/key-token/key-wbtc-59760-usda.clar

```diff
@@ -26,55 +34,47 @@
 )
 
 (define-read-only (get-decimals)
-  (ok u8)
+  (ok u6)
 )
 
 (define-read-only (get-balance (account principal))
   (ok (ft-get-balance key-wbtc-59760-usda account))
 )
 
-(define-read-only (get-total-supply)
-  (ok (ft-get-supply key-wbtc-59760-usda))
-)
-
 (define-public (set-token-uri (value (string-utf8 256)))
-  ;; TODO : Authorization Check
-  ;;(if (is-eq tx-sender (contract-call? .OWNER))
+  (begin
+    (asserts! (is-eq tx-sender (var-get contract-owner)) ERR-NOT-AUTHORIZED)
     (ok (var-set token-uri value))
-  ;;  (err ERR-NOT-AUTHORIZED)
-  ;;)
+  )
 )
 
 (define-read-only (get-token-uri)
   (ok (some (var-get token-uri)))
 )
 
 (define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
-  (match (ft-transfer? key-wbtc-59760-usda amount sender recipient)
-    response (begin
-      (print memo)
-      (ok response)
+  (begin
+    (asserts! (is-eq sender tx-sender) ERR-NOT-AUTHORIZED)
+    (match (ft-transfer? key-wbtc-59760-usda amount sender recipient)
+      response (begin
+        (print memo)
+        (ok response)
+      )
+      error (err error)
     )
-    error (err error)
   )
 )
 
-;; ---------------------------------------------------------
-;; ayUSDA token trait
-;; ---------------------------------------------------------
-
-;; Mint method for key-wbtc-59760-usda
 (define-public (mint (recipient principal) (amount uint))
   (begin
-    ;;(asserts! (is-eq contract-caller .collateral-rebalancing-pool) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq contract-caller (var-get contract-owner)) ERR-NOT-AUTHORIZED)
```

Comment:
Why `contract-caller` and not `tx-sender`? Using `tx-sender` is more flexible but perhaps there is a very strict contract chain in this instance. Seems inconsistent considering the guard for `set-token-uri` checks against `tx-sender`.

Reply:
I agree that `tx-sender` is more flexible, but do think we want to be as restrictive as possible on something like access control, so my preference is to use `contract-caller` wherever possible.
In this particular case, because the only contract that would mint `key-tokens` is `collateral-rebalancing-pool` (as part of `add-to-position`, and no other use-cases, so `contract-caller` makes sense.

Agreed on the inconsistency - need fixed.